braintree/braintree_ios
# Fix `ConfigurationCallbackStorage` Crash

### Summary of changes

- Fixes https://github.com/braintree/braintree_ios/issues/1382
- Remove `ConfigurationCallbackStorage`
- Add `async/throws` `get` method
- Update `getConfig()` to use `async/throws` - use task block
- Update tests to use `async/await` methods

### To Test
- The configuration is being cached and we use the cache if available - verify via FPTI
- When triggering a memory warning a crash is not present
- There are no crashes when instantiating multiple instances at the same time

### Checklist


### Authors
@scannillo @jaxdesmarais @KunJeongPark
